pkg/services/dns/dns.go

```diff
@@ -168,10 +168,23 @@ func (s *Server) Mux() http.Handler {
 			return
 		}
 
-		s.handler.zonesLock.Lock()
-		s.handler.zones = append([]types.Zone{req}, s.handler.zones...)
-		s.handler.zonesLock.Unlock()
+		s.addZone(req)
 		w.WriteHeader(http.StatusOK)
 	})
 	return mux
 }
+
+func (s *Server) addZone(req types.Zone) {
+	s.handler.zonesLock.Lock()
+	for i, zone := range s.handler.zones {
+		if zone.Name == req.Name {
+			req.Records = append(req.Records, zone.Records...)
+			merge := append([]types.Zone{req}, s.handler.zones[i+1:]...)
+			s.handler.zones = append(s.handler.zones[:i], merge...)
+			s.handler.zonesLock.Unlock()
+			return
+		}
+	}
```

Comment:
Worth a small comment `// No existing zone for req.Name, add new one`

Reply:
Done this change.